### PR TITLE
CroSimulator jax.grad

### PR DESCRIPTION
**Pull request by xzackli**
CroSimulator no longer inherits SimulatorBase. Instead, it is created with arbitrary defaults, and you can take gradients through simulate(). 

Defaults come from the constructor, which precomputes observation-fixed quantities (the rotation phases). This makes jax.jit and jax.grad work on any subset of inputs for simulate(), unspecified args are closed-over constants.

### CODE REVIEW

**Comment by xzackli** [comment]
Arguments you pass to `simulate()` are differentiated through; arguments you omit are closed-over constants from `self`. Example usage:

```python
import jax
import jax.numpy as jnp
import lusee

# --- Setup (unchanged) ---
sim = lusee.CroSimulator(obs, beams, sky, Tground=0.0,
                         combinations=[(0,0)], freq=freq, lmax=32)

# --- Forward pass (no args = use defaults from constructor) ---
wf = sim.simulate()   # shape (Ntimes, Nchannels, Nfreq)

# --- Gradient w.r.t. beam ALMs (sky held constant via closure) ---
@jax.jit
def loss_beam(beam_alms):
    return jnp.sum(sim.simulate(beam_alms) ** 2)

grad_beam = jax.grad(loss_beam)(sim.beam_alms)

# --- Gradient w.r.t. sky (beam held constant) ---
@jax.jit
def loss_sky(sky_mepa):
    return jnp.sum(sim.simulate(sky_mepa=sky_mepa) ** 2)

grad_sky = jax.grad(loss_sky)(sim.sky_mepa)

# --- Gradient w.r.t. beam and sky simultaneously ---
@jax.jit
def loss_both(beam_and_sky):
    b, s = beam_and_sky
    return jnp.sum(sim.simulate(b, s) ** 2)

grad_b, grad_s = jax.grad(loss_both)((sim.beam_alms, sim.sky_mepa))

# --- Gradient w.r.t. Tground ---
gp = jnp.ones_like(sim.ground_power) * 0.1
grad_T = jax.grad(lambda T: jnp.sum(
    sim.simulate(Tground=T, ground_power=gp) ** 2))(jnp.array(200.0))
```
